homeassistant/components/bluetooth/update_coordinator.py

```diff
@@ -84,17 +112,9 @@ def _async_stop(self) -> None:
             self._cancel_track_unavailable = None
 
     @callback
-    def _async_handle_unavailable(self, address: str) -> None:
-        """Handle the device going unavailable."""
-        self._present = False
-
-    @callback
-    def _async_handle_bluetooth_event(
-        self,
-        service_info: BluetoothServiceInfoBleak,
-        change: BluetoothChange,
+    def _async_handle_unavailable(
+        self, service_info: BluetoothServiceInfoBleak
     ) -> None:
-        """Handle a Bluetooth event."""
-        self.last_seen = time.monotonic()
-        self.name = service_info.name
-        self._present = True
```

Comment:
This is also a problem in switchbot since we don't get here and set the device as present when the adv data hasn't changed

https://github.com/home-assistant/core/blob/635ed562eebe7919e6edfee3c22a70b4ce33e7d0/homeassistant/components/switchbot/coordinator.py#L80